core/deployment/src/main/java/io/quarkus/deployment/builditem/MainBytecodeRecorderBuildItem.java

```diff
@@ -6,13 +6,23 @@
 public final class MainBytecodeRecorderBuildItem extends MultiBuildItem {
 
     private final BytecodeRecorderImpl bytecodeRecorder;
+    private final String generatedStartupContextClassName;
```

Comment:
The reason I did it this way is because the `MainClassBuildStep` can continue to just consume `List<MainBytecodeRecorderBuildItem>` and the order is properly worked out.

Reply:
I'm still not loving this, but the Monday brain fails to come up with any better ideas, so... I guess it's fine.

Reply:
Same here :)